org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/codemining/JavaMethodParameterCodeMining.java

```diff
@@ -40,6 +41,17 @@ public JavaMethodParameterCodeMining(Expression parameterNode, int parameterInde
 		}
 	}
 
+	public JavaMethodParameterCodeMining(Expression parameterNode, int parameterIndex, IMethodBinding mbinding, boolean isVarargs, ICodeMiningProvider provider) {
```

Comment:
The constructor is the exact same code except for `binding.getParameterNames()`. Please extract the common code either to a method or to the call sites of the constructor.